src/ddprof_context_lib.cc

```diff
@@ -42,213 +42,104 @@ static void order_watchers(ddprof::span<PerfWatcher> watchers) {
       });
 }
 
-/****************************  Argument Processor  ***************************/
-DDRes context_set(DDProfInput *input, DDProfContext *ctx) {
-  *ctx = {};
-  setup_logger(input->log_mode, input->log_level);
+namespace {
+void copy_cli_values(const DDProfCLI &ddprof_cli, DDProfContext &ctx) {
+
```

Comment:
This is not perfect for now, as it requires an extra copy step. We could extract underlying structure to make copies transparent.